api/application/applications.go

```diff
@@ -8,139 +8,114 @@ import (
 )
 
 //WIP
-func Fetch(network *net.Network) string {
+func Fetch(network *net.Network) (string, error) {
 	url := "/v1/applications/fetch"
 	body, err := network.SendGetRequest(url)
 	if err != nil {
-		fmt.Println(err)
+		return string(body), err
 	}
 	// TODO return model
-	return string(body)
+	return string(body), nil
 }
 
-func Applications(network *net.Network) []models.ApplicationSummary {
+func Applications(network *net.Network) ([]models.ApplicationSummary, error) {
 	url := fmt.Sprintf("/v1/applications")
-	body, err := network.SendGetRequest(url)
-	if err != nil {
-		fmt.Println(err)
-	}
+    var appSummary []models.ApplicationSummary
+    body, err := network.SendGetRequest(url)
+    if err != nil {
+        return appSummary, err
+    }
 
-	var appSummary []models.ApplicationSummary
 	err = json.Unmarshal(body, &appSummary)
-	if err != nil {
-		fmt.Println(err)
-	}
-	return appSummary
+	return appSummary, err
 }
 
-func Create(network *net.Network, filePath string) models.TaskSummary {
+func Create(network *net.Network, filePath string) (models.TaskSummary, error) {
 	url := "/v1/applications"
-	body, err := network.SendPostFileRequest(url, filePath)
-	if err != nil {
-		fmt.Println(err)
-	}
-	var response models.TaskSummary
+    var response models.TaskSummary
+    body, err := network.SendPostFileRequest(url, filePath)
+    if err != nil {
+        return response, err
+    }
 	err = json.Unmarshal(body, &response)
-	if err != nil {
-		fmt.Println(err)
-	}
-	return response
+	return response, err
 }
 
-func CreateFromBytes(network *net.Network, blueprint []byte)  models.TaskSummary {
+func CreateFromBytes(network *net.Network, blueprint []byte) (models.TaskSummary, error) {
 	url := "/v1/applications"
-	
-	body, err := network.SendPostRequest(url, blueprint)
-	if err != nil {
-		fmt.Println(err)
-	}
-	var response models.TaskSummary
+    var response models.TaskSummary
+    body, err := network.SendPostRequest(url, blueprint)
+    if err != nil {
+        return response, err
+    }
 	err = json.Unmarshal(body, &response)
-	if err != nil {
-		fmt.Println(err)
-	}
-	return response
+	return response, err
 }
 
 // WIP
-func Descendants(network *net.Network, app string) string {
+func Descendants(network *net.Network, app string) (string, error) {
 	url := fmt.Sprintf("/v1/applications/%s/descendants", app) 
 	
 	body, err := network.SendGetRequest(url)
-	if err != nil {
-		fmt.Println(err)
-	}
 	// TODO return model
-	return string(body)
+	return string(body), err
 }
 
 // WIP
-func DescendantsSensor(network *net.Network, app, sensor string) string {
+func DescendantsSensor(network *net.Network, app, sensor string) (string, error) {
 	url := fmt.Sprintf("/v1/applications/%s/descendants/sensor/%s", app, sensor) 
 	
 	body, err := network.SendGetRequest(url)
-	if err != nil {
-		fmt.Println(err)
-	}
 	// TODO return model
-	return string(body)
+	return string(body), err
```

Comment:
Need to be careful here, as it is likely that string var body will not have been created. Safer to just return empty string `""`.